### PR TITLE
Add Logger functions to GYResources

### PR DESCRIPTION
**Pull request by gvNN7**
- [x] Add LoggerController #74 
- [x] Add log to functions #69 
- [ ] Add logger test

### CODE REVIEW

**Comment by WendelHime** [comment]
Please, fix conflicts

**Comment by WendelHime** [comment]
fix also https://landscape.io/github/CapivaraProjects/GYResources/pulls/83 problems detected

File: api/gyresources/endpoints/PlantController.py [hunk 1]
```diff
@@ -155,6 +171,13 @@ def put(self):
                 response=plant,
                 message="Plant sucessfuly updated.",
                 status=204), 200
+        str(message.__dict__)
```
Comment (WendelHime) [inline api/gyresources/endpoints/PlantController.py]: Should be:
```python3
message = str(message.__dict__)
```
Replace it in all files please

File: api/gyresources/endpoints/DiseaseController.py [hunk 2]
```diff
@@ -79,6 +86,12 @@ def get(self):
                 for content in result['content']:
                     content.plant = content.plant.__dict__
                     response.append(content)
+                Logger.Logger.create(flask_app.config["ELASTICURL"],
+                                     'Informative',
+                                     'Ok',
+                                     'get()',
+                                     str(response.__dict__),
```
Comment (WendelHime) [inline api/gyresources/endpoints/DiseaseController.py]: response is a list, you cannot convert list to dict. <br />
So, use `str(response)` instead `str(response.__dict__)`

Comment (gvNN7) [reply api/gyresources/endpoints/DiseaseController.py]: fixed

File: api/gyresources/endpoints/UserController.py [hunk 3]
```diff
@@ -201,24 +255,48 @@ def delete(self):
         try:
             status = repository.delete(user)
             if (status):
+                Logger.Logger.create(flask_app.config["ELASTICURL"],
+                                     'Informative',
+                                     'User deleted sucessfuly',
+                                     'delete()',
+                                     str(user.__dict__),
```
Comment (WendelHime) [inline api/gyresources/endpoints/UserController.py]: change `user.__dict__` to `status`

File: api/gyresources/endpoints/TypeController.py [hunk 4]
```diff
@@ -190,23 +244,46 @@ def delete(self):
         try:
             status = repository.delete(type)
             if (status):
+                Logger.Logger.create(flask_app.config["ELASTICURL"],
+                                     'Informative',
+                                     'Type deleted sucessfuly',
+                                     'post()',
+                                     str(type.__dict__),
```
Comment (WendelHime) [inline api/gyresources/endpoints/TypeController.py]: change `type.__dict__` to `status`

Comment (gvNN7) [reply api/gyresources/endpoints/TypeController.py]: files changed

File: tests/test_log.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+                                   obs="testObs",
+                                   config="Test")
+
+@pytest.mark.order1
+def test_log(generic_log=generic_log):
+    resp = client().post('/api/gyresources/logs/', data=str(
+        json.dumps(generic_log.__dict__)), headers={
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'})
+    assert resp.status_code == 200
```
Comment (WendelHime) [inline tests/test_log.py]: resp should be:
```python3
resp = json.loads(resp.get_data(as_text=True))
resp = namedtuple("BaseResponse", resp.keys())(*resp.values())
assert resp.status_code == 201
```

File: api/gyresources/endpoints/LoggerController.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        """
+        Method used to create the log from the action
+        """
+        try:
+            Logger.create(url=flask_app.config["ELASTICURL"],
+                          type=request.args.get('type'),
+                          message=request.args.get('message'),
+                          function=request.args.get('function'),
+                          obs=request.args.get('message'),
+                          config=request.args.get('message'))
```
Comment (WendelHime) [inline api/gyresources/endpoints/LoggerController.py]: fix fields in request.args.get('') your're setting obs, config as message

File: tests/test_log.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+@pytest.mark.order2
+def test_log_emptyRequiredFields(generic_log=generic_log):
+    data = {
+            "type": "",
+            "function": "",
+            "config": ""
+            }
+    resp = client().post('/api/gyresources/logs/', data=str(
+        json.dumps(generic_log.__dict__)), headers={
```
Comment (WendelHime) [inline tests/test_log.py]: change generic_log.__dict__ to data

File: tests/test_log.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+@pytest.mark.order3
+def test_log_incorrectFields(generic_log=generic_log):
+    data = {
+            "type": "",
+            "function": "null",
+            "config": "unknown"
+            }
+    resp = client().post('/api/gyresources/logs/', data=str(
+        json.dumps(generic_log.__dict__)), headers={
```
Comment (WendelHime) [inline tests/test_log.py]: same thing here

**Pull request merged**